prime-router/src/test/kotlin/history/azure/DeliveryFunctionTests.kt

```diff
@@ -1555,6 +1568,422 @@ class DeliveryFunctionTests : Logging {
+
+            val response = DeliveryFunction().getDeliveriesHistory(httpRequestMessage, receiver1.fullName)
+            assertThat(response.status).isEqualTo(HttpStatus.OK)
+        }
+
+        @Test
+        fun `test successfully returns data for a prime admin`() {
+            val httpRequestMessage = MockHttpRequestMessage(
+                """
+                {
+                    "sort": {
+                        "direction": "DESC",
+                        "property": "test_result_count"
+                    },
+                    "pagination": {
+                        "page": 1,
+                        "limit": 100
+                    },
+                    "filters": [
+                    ]
+                }
+                """.trimIndent()
+            )
+
+            val jwt = mapOf("organization" to listOf(oktaSystemAdminGroup), "sub" to "test@cdc.gov")
+            val claims = AuthenticatedClaims(jwt, AuthenticationType.Okta)
+
+            mockkObject(AuthenticatedClaims)
+            every { AuthenticatedClaims.Companion.authenticate(any()) } returns claims
+            mockkConstructor(DeliveryHistoryDatabaseAccess::class)
+            every {
+                anyConstructed<DeliveryHistoryDatabaseAccess>().getDeliveries(
+                    any(), organization1.name, null, null, null, null
+                )
+            } returns ApiSearchResult(10, 0, emptyList())
+
+            val response = DeliveryFunction().getDeliveriesHistory(httpRequestMessage, receiver1.fullName)
+            assertThat(response.status).isEqualTo(HttpStatus.OK)
+        }
+
+        @Test
+        fun `test successfully returns an API response`() {
+            val httpRequestMessage = MockHttpRequestMessage(
+                """
+                {
+                    "sort": {
+                        "direction": "DESC",
+                        "property": "test_result_count"
+                    },
+                    "pagination": {
+                        "page": 2,
+                        "limit": 10
+                    },
+                    "filters": [
+                        {
+                            "value": "2023-05-21T00:00:00+00:00",
+                            "filterName": "UNTIL"
+                        }
+                    ]
+                }
+                """.trimIndent()
+            )
+
+            val jwt = mapOf("organization" to listOf(oktaSystemAdminGroup), "sub" to "test@cdc.gov")
+            val claims = AuthenticatedClaims(jwt, AuthenticationType.Okta)
+
+            mockkObject(AuthenticatedClaims)
+            every { AuthenticatedClaims.Companion.authenticate(any()) } returns claims
+            mockkConstructor(DeliveryHistoryDatabaseAccess::class)
+            every {
+                anyConstructed<DeliveryHistoryDatabaseAccess>().getDeliveries(
+                    any(),
+                    organization1.name, null, null, null, null
```

Comment:
It doesn't look like this matches what the mock is getting called with (looks like receiver parameter is getting passed).

After that and fixing the assertion on 1764L the test passed for me